include/aspect/material_model/interface.h

```diff
@@ -217,6 +217,16 @@ namespace aspect
                           const bool use_strain_rates = true);
 
       /**
+       * Copy Constructor.
+       */
+      MaterialModelInputs(const MaterialModelInputs<dim> &material);
+
+      /**
+       * Assignment operator.
+       */
+      MaterialModelInputs &operator=(const MaterialModelInputs &material);
```

Comment:
@jdannberg, I am fine with changing it to the `MaterialModelInputs<dim> in (base_inputs)` style in the Newton tests. I indeed have some more tests which are not part of the repository  yet, but changing it should not be much work.